test-network-function/lifecycle/suite.go

```diff
@@ -113,95 +113,111 @@ var _ = ginkgo.Describe(common.LifecycleTestKey, func() {
 	var elapsed time.Duration
-	var notReadyDeployments []string
+	var notReadyPodSets []string
 
 	for elapsed < timeout {
-		_, notReadyDeployments = getDeployments(namespace)
-		log.Debugf("Waiting for deployments to get ready, remaining: %d deployments", len(notReadyDeployments))
-		if len(notReadyDeployments) == 0 {
+		_, notReadyPodSets = GetPodSets(namespace, resourceType)
+		log.Debugf("Waiting for %s to get ready, remaining: %d PodSets", string(resourceType), len(notReadyPodSets))
+		if len(notReadyPodSets) == 0 {
 			break
 		}
 		time.Sleep(pollingPeriod)
 		elapsed += pollingPeriod
 	}
-	return len(notReadyDeployments)
+	return len(notReadyPodSets)
 }
 
 // restoreDeployments is the last attempt to restore the original test deployments' replicaCount
 func restoreDeployments(env *config.TestEnvironment) {
-	for _, deployment := range env.DeploymentsUnderTest {
+	for i := range env.DeploymentsUnderTest {
 		// For each test deployment in the namespace, refresh the current replicas and compare.
-		deployments, notReadyDeployments := getDeployments(deployment.Namespace)
-
-		if len(notReadyDeployments) > 0 {
-			// Wait until the deployment is ready
-			notReady := waitForAllDeploymentsReady(deployment.Namespace, scalingTimeout, scalingPollingPeriod)
-			if notReady != 0 {
-				collectNodeAndPendingPodInfo(deployment.Namespace)
-				log.Fatalf("Could not restore deployment replicaCount for namespace %s.", deployment.Namespace)
-			}
-		}
+		refreshReplicas(&env.DeploymentsUnderTest[i], env)
+	}
+}
+
+// restoreStateFullSet is the last attempt to restore the original test PodSets' replicaCount
+func restoreStateFullSet(env *config.TestEnvironment) {
+	for i := range env.StateFullSetUnderTest {
+		// For each test StateFullSet in the namespace, refresh the current replicas and compare.
+		refreshReplicas(&env.StateFullSetUnderTest[i], env)
+	}
+}
 
-		if deployment.Hpa.HpaName != "" { // it have hpa and need to update the max min
-			runHpaScalingTest(deployment)
+func refreshReplicas(podset *configsections.PodSet, env *config.TestEnvironment) {
+	podsets, notReadyPodsets := GetPodSets(podset.Namespace, podset.Type)
+
+	if len(notReadyPodsets) > 0 {
+		// Wait until the deployment/replicaset is ready
+		notReady := waitForAllPodSetsReady(podset.Namespace, scalingTimeout, scalingPollingPeriod, podset.Type)
+		if notReady != 0 {
+			collectNodeAndPendingPodInfo(podset.Namespace)
+			log.Fatalf("Could not restore %s replicaCount for namespace %s.", string(podset.Type), podset.Namespace)
 		}
-		if deployments[deployment.Name].Replicas != deployment.Replicas {
-			log.Warn("Deployment ", deployment.Name, " replicaCount (", deployment.Replicas, ") needs to be restored.")
+	}
+	if podset.Hpa.HpaName != "" { // it have hpa and need to update the max min
+		runHpaScalingTest(podset)
+	}
```

Comment:
missing else?

Reply:
I think it's ok, if there's no HpaName there's no need to run any HPA scaling test.